Modules/_posixsubprocess.c

```diff
@@ -506,6 +576,13 @@ child_exec(char *const exec_array[],
     if (restore_signals)
         _Py_RestoreSignals();
 
+#ifdef VFORK_USABLE
+    if (child_sigmask) {
+        reset_signal_handlers(child_sigmask);
+        pthread_sigmask(SIG_SETMASK, child_sigmask, NULL);
```

Comment:
I added an error check on this one because it is doable in this spot even though unlikely.
I also verified that reset_signal_handlers has no need to report any errors.